src/Parsers/ASTSelectQuery.h

```diff
@@ -143,6 +142,9 @@ class ASTSelectQuery : public IAST
 
     QueryKind getQueryKind() const override { return QueryKind::Select; }
 
+    mutable std::optional<bool> has_query_parameters;
```

Comment:
this field can be made private? also let's add a comment why it is `std::optional`

Reply:
addressed it.